src/GraphQL.Tests/Utilities/Visitors/DeprecatedTypeReferenceVisitorTests.cs

```diff
@@ -0,0 +1,273 @@
+            Name = "DeprecatedFilterInput",
+            DeprecationReason = "Use NewFilterInput instead"
+        };
+        deprecatedInputType.Field<StringGraphType>("term");
+
+        var queryType = new ObjectGraphType { Name = "Query" };
+        queryType.Field<StringGraphType>("search")
+            .Argument(deprecatedInputType, "filter")
+            .Resolve(ctx => "result");
+
+        var schema = new Schema { Query = queryType };
+        schema.RegisterType(deprecatedInputType);
+        schema.RegisterVisitor(new DeprecatedTypeReferenceVisitor());
+
+        // Act & Assert
+        var exception = Should.Throw<InvalidOperationException>(() => schema.Initialize());
+        exception.Message.ShouldBe("Non-deprecated argument 'Query.search.filter' references deprecated type 'DeprecatedFilterInput'.");
+    }
+
+    [Fact]
+    public void Should_Throw_When_InterfaceFieldArgument_References_Deprecated_Type()
+    {
+        // Arrange
+        var deprecatedInputType = new InputObjectGraphType
+        {
+            Name = "DeprecatedSortInput",
+            DeprecationReason = "Use NewSortInput instead"
+        };
+        deprecatedInputType.Field<StringGraphType>("field");
+
+        var activeInterface = new InterfaceGraphType { Name = "ISearchable" };
+        activeInterface.Field<StringGraphType>("find")
+            .Argument(deprecatedInputType, "sort")
+            .Resolve(ctx => "result");
+
+        var schema = new Schema { Query = new ObjectGraphType { Name = "Query" } };
+        schema.RegisterType(deprecatedInputType);
+        schema.RegisterType(activeInterface);
+        schema.RegisterVisitor(new DeprecatedTypeReferenceVisitor());
+
+        // Act & Assert
+        var exception = Should.Throw<InvalidOperationException>(() => schema.Initialize());
+        exception.Message.ShouldBe("Non-deprecated argument 'ISearchable.find.sort' references deprecated type 'DeprecatedSortInput'.");
+    }
+
+    [Fact]
+    public void Should_Handle_Wrapped_Deprecated_Types_In_ObjectFields()
+    {
+        // Arrange
+        var deprecatedType = new ObjectGraphType
+        {
+            Name = "DeprecatedUser",
+            DeprecationReason = "Use NewUser instead"
+        };
+        deprecatedType.Field<StringGraphType>("id");
+
+        var activeType = new ObjectGraphType { Name = "Post" };
+        activeType.Field<NonNullGraphType<ListGraphType<NonNullGraphType<ObjectGraphType>>>>("authors")
+            .Type(new NonNullGraphType(new ListGraphType(new NonNullGraphType(deprecatedType))));
+
+        var schema = new Schema { Query = activeType };
+        schema.RegisterType(deprecatedType);
+        schema.RegisterVisitor(new DeprecatedTypeReferenceVisitor());
+
+        // Act & Assert
+        var exception = Should.Throw<InvalidOperationException>(() => schema.Initialize());
+        exception.Message.ShouldBe("Non-deprecated field 'Post.authors' references deprecated type 'DeprecatedUser'.");
+    }
+}
```

Comment:
What happens when all type fields are deprecated? Should the type be marked as deprecated too?